sound/soc/sof/intel/cnl.c

```diff
@@ -82,9 +82,22 @@ irqreturn_t cnl_ipc_irq_thread(int irq, void *context)
 			 msg, msg_ext);
 
 		/* handle messages from DSP */
-		if ((hipctdr & SOF_IPC_PANIC_MAGIC_MASK) ==
-		   SOF_IPC_PANIC_MAGIC) {
-			snd_sof_dsp_panic(sdev, HDA_DSP_PANIC_OFFSET(msg_ext));
+		if ((hipctdr & SOF_IPC_PANIC_MAGIC_MASK) == SOF_IPC_PANIC_MAGIC) {
+			struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+			bool no_recovery = true;
```

Comment:
Or we could use 'last_boot_attempt'?

This all comes from my (bad?) choice of the `bool non_recoverable` new parameter to snd_sof_dsp_panic()...
The aim was to have a **true** passed in case this is a 'real' and 'definite' panic and a **false** would indicate that the platform code might be able to deal with it, but this is still a panic from the firmware and if user wants to see _all_ dumps then it should be shown.

Reply:
Hrm, `last_boot_attempt` is true for the period when we are booting up but after that if the panic comes when we have already booted up, it is a bit strange.
Probably the `non_recoverable` is a bit better match for the cases we handle here.